lib/core/src/text.rs

```diff
@@ -75,29 +76,66 @@ impl TextComponent {
+        self.buffers.refresh(&self.gl_context,self.content.refresh_info(fonts));
+
+        if self.msdf_texture_rows != fonts.get_render_info(self.content.font).msdf_texture.rows() {
+            self.update_msdf_texture(fonts);
+        }
+
         let gl_context     = &self.gl_context;
         let vertices_count = self.buffers.vertices_count() as i32;
-        let lines          = &mut self.lines;
-        let font           = fonts.get_render_info(self.font);
-        let content_ref    = ContentRef{lines,font};
 
-        self.buffers.refresh(gl_context,content_ref);
         gl_context.use_program(Some(&self.gl_program));
-        self.update_uniforms();
+        self.update_uniforms(fonts);
         self.bind_buffer_to_attribute("position",&self.buffers.vertex_position);
         self.bind_buffer_to_attribute("tex_coord",&self.buffers.texture_coords);
         self.setup_blending();
         gl_context.bind_texture(Context::TEXTURE_2D, Some(&self.gl_msdf_texture));
         gl_context.draw_arrays(WebGl2RenderingContext::TRIANGLES,0,vertices_count);
     }
 
-    fn update_uniforms(&self) {
+    fn update_uniforms(&self, fonts:&mut Fonts) {
         let gl_context          = &self.gl_context;
         let to_scene            = self.to_scene_matrix();
         let to_scene_ref        = to_scene.as_ref();
+        let msdf_width          = MsdfTexture::WIDTH as f32;
+        let msdf_height         = fonts.get_render_info(self.content.font).msdf_texture.rows() as f32;
         let to_scene_loc        = gl_context.get_uniform_location(&self.gl_program,"to_scene");
+        let msdf_size_loc       = gl_context.get_uniform_location(&self.gl_program,"msdf_size");
         let transpose           = false;
         gl_context.uniform_matrix3fv_with_f32_array(to_scene_loc.as_ref(),transpose,to_scene_ref);
+        gl_context.uniform2f(msdf_size_loc.as_ref(),msdf_width,msdf_height);
+
+    }
+
+    fn update_msdf_texture(&mut self, fonts:&mut Fonts) {
+        let gl_context   = &self.gl_context;
+        let font_msdf    = &fonts.get_render_info(self.content.font).msdf_texture;
+        let target       = Context::TEXTURE_2D;
+        let width        = MsdfTexture::WIDTH as i32;
+        let height       = font_msdf.rows() as i32;
+        let border       = 0;
+        let tex_level    = 0;
+        let format       = Context::RGB;
+        let internal_fmt = Context::RGB as i32;
+        let tex_type     = Context::UNSIGNED_BYTE;
+        let data         = Some(font_msdf.data.as_slice());
+
+        gl_context.bind_texture(target,Some(&self.gl_msdf_texture));
+        let tex_image_result =
+            gl_context.tex_image_2d_with_i32_and_i32_and_i32_and_format_and_type_and_opt_u8_array
+            ( target
+            , tex_level
+            , internal_fmt
+            , width
+            , height
+            , border
+            , format
+            , tex_type
+            , data
+            );
```

Comment:
I'm just wondering if we really want such code to occupy 11 lines. I mean, is it more readable this way?

Reply:
For me It's better to be one-two lines, the current shape came from https://github.com/luna/basegl/pull/62#discussion_r353527495

Reply:
Ok, I see, that in times of that comment we had 80 chars limit.